simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/component/ocppcommctrlr/NetworkConfigurationPriorityVariableAccessor.java

```diff
@@ -0,0 +1,126 @@
+    private final Map<AttributeType, VariableGetter> variableGetters = ImmutableMap.<AttributeType, VariableGetter>builder()
+            .put(AttributeType.ACTUAL, this::getActualValue)
+            .build();
+
+    private final Map<AttributeType, SetVariableValidator> variableValidators = ImmutableMap.<AttributeType, SetVariableValidator>builder()
+            .put(AttributeType.ACTUAL, this::validateActualValue)
+            .build();
+
+    private final Map<AttributeType, VariableSetter> variableSetters = ImmutableMap.<AttributeType, VariableSetter>builder()
+            .put(AttributeType.ACTUAL, this::setActualValue)
+            .build();
+
+    @Override
+    public String getVariableName() {
+        return NAME;
+    }
+
+    @Override
+    public Map<AttributeType, VariableGetter> getVariableGetters() {
+        return variableGetters;
+    }
+
+    @Override
+    public Map<AttributeType, VariableSetter> getVariableSetters() {
+        return variableSetters;
+    }
+
+    @Override
+    public Map<AttributeType, SetVariableValidator> getVariableValidators() {
+        return variableValidators;
+    }
+
+    @Override
+    public List<ReportDatum> generateReportData(String componentName) {
+        Component component = new Component()
+                .withName(new CiString.CiString50(componentName));
+
+        VariableAttribute variableAttribute = new VariableAttribute()
+                .withValue(new CiString.CiString1000(String.valueOf(getStationStore().getNetworkConfigurationPriority())))
+                .withPersistence(false)
+                .withConstant(false);
+
+        VariableCharacteristics variableCharacteristics = new VariableCharacteristics()
+                .withDataType(SEQUENCE_LIST)
+                .withSupportsMonitoring(false);
+
+        ReportDatum reportDatum = new ReportDatum()
+                .withComponent(component)
+                .withVariable(new Variable().withName(new CiString.CiString50(NAME)))
+                .withVariableCharacteristics(variableCharacteristics)
+                .withVariableAttribute(singletonList(variableAttribute));
+
+        return singletonList(reportDatum);
+    }
+
+    @Override
+    public boolean isMutable() {
+        return true;
+    }
+
+    private SetVariableResult validateActualValue(AttributePath attributePath, CiString.CiString1000 attributeValue) {
+        SetVariableResult setVariableResult = new SetVariableResult()
+                .withComponent(attributePath.getComponent())
+                .withVariable(attributePath.getVariable())
+                .withAttributeType(SetVariableResult.AttributeType.fromValue(attributePath.getAttributeType().getName()));
+
+        if (!isNumeric(attributeValue.toString())) {
+            return setVariableResult.withAttributeStatus(SetVariableResult.AttributeStatus.INVALID_VALUE);
+        } else if (!getStationStore().getNetworkConnectionProfiles().containsKey(Integer.parseInt(attributeValue.toString()))) {
```

Comment:
all this nested If's are difficult to read and understand what is going on and if there is a connection between branches 
it would be easier to read if the code will be organised in follow way 
```
if (!isNumeric(attributeValue.toString())) {
    return setVariableResult.withAttributeStatus(SetVariableResult.AttributeStatus.INVALID_VALUE);
} 
if (!getStationStore().getNetworkConnectionProfiles().containsKey(Integer.parseInt(attributeValue.toString()))) {
    return setVariableResult.withAttributeStatus(SetVariableResult.AttributeStatus.INVALID_VALUE);
}

return setVariableResult.withAttributeStatus(SetVariableResult.AttributeStatus.ACCEPTED);
```